### ISSUE TITLE
Enable the OpenSSF Scorecard Github Action

### ISSUE THREAD

**Issue by joycebrum** [google-github-actions/auth#263]
### TL;DR

Enable [OpenSSF Scorecard Github Action][sc-gha] on repo to improve the project's supply chain security posture.

[sc-gha]: https://github.com/ossf/scorecard-action

### Detailed design

Hi, I am talking on behalf of Google and the [OpenSSF][ossf] and I'm working on helping open source projects to increase their Supply Chain Security Posture.

To help on that, I would like to suggest the [OpenSSF Scorecard Action][sc-gha] that automatically runs on the repository checking for possible security improvements that can be made to avoid some known supply-chain security threats.

The auth project already scored amazing well on scorecard checks, having only a few improvements that could be done such as Token-Permissions and CI-Tests. The first one seems to be specially applicable to auth since it mitigates some threats on the workflow actions.

To more informations about the Scorecard Checks, please see the [Scorecard Documentation][checks].

Let me know if you are interested in a PR configuring the OpenSSF Scorecard Action and, optionally, its [badge][badge]. 

In case of doubts or concerns please feel free to reach out to me.

[badge]: https://openssf.org/blog/2022/09/08/show-off-your-security-score-announcing-scorecards-badges/
[ossf]: https://openssf.org/
[sc-gha]: https://github.com/ossf/scorecard-action
[checks]: https://github.com/ossf/scorecard/blob/main/docs/checks.md#check-documentation

### Additional information

Additional Reading:

- [Reducing security risk in open source software with GitHub Actions and OpenSSF Scorecards V4](https://github.blog/2022-01-19-reducing-security-risk-oss-actions-opensff-scorecards-v4/)

**Comment by joycebrum** [comment]
Hi @sethvargo thanks for the feedback. Code Review and CI-Tests really may be false positives due to the bot activity, but I believe you don't need to worry about the Signed-Releases since it should not be included in the Security Dashboard since it is returned as -1 (Not Applicable) and it is not considered to calculate the final Scorecard score.

For the token permission, the actions might have write permissions, it just need to be granted on the job level instead of the top level or by the default.

It is a good practice to set minimal permissions to github workflows (see the section "Use credentials that are minimally scoped" in [github documentation](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions) about it).

Example using the https://github.com/google-github-actions/auth/blob/main/.github/workflows/release.yml action:

```
name: 'Release'

on:
  push:
    branches:
      - 'main'

permissions: 
  contents: read

jobs:
  release:
    permissions:
      contents: write # if it creates the release on github 
      id-token: write # to enable OIDC token
    if: "startsWith(github.event.head_commit.message, 'Release: v')"
    name: 'Release'
    uses: 'google-github-actions/.github/.github/workflows/release.yml@v0'
```

This would be considered a 10/10 github workflow in Token Permissions check. I can suggest these changes if you rather, they are quite simple to apply in most of the cases.

The OpenSSF Scorecard Action is basically a way to be notified of future "human errors" such as adding a new workflow without minimal permissions or hash-pinning dependencies, but all dashboard alert can be dismissed (and AFAIK won't appear anymore on future runs) by selecting any of the options below:

![image](https://user-images.githubusercontent.com/22223372/232844814-0b76ce78-6884-4bba-a42b-93a2f19c5448.png)

Let me know if I can help in anything!
